### PR TITLE
Simplifying acceptance step

### PR DESCRIPTION
**Pull request by dfm**
This should be identical but with less code.

cc @eigenfoo

### CODE REVIEW

**Comment by eigenfoo** [comment]
A similar edit should probably be made to the `_slow` version too? This is an awful lot of blog posts that @ColCarroll is subjecting us to.

File: minimc/minimc.py [hunk 1]
```diff
@@ -75,19 +75,16 @@ def hamiltonian_monte_carlo(
...
 
-        if abs(energy_change) < max_energy_change:
-            # Check Metropolis acceptance criterion
-            if np.random.rand() < p_accept:
-                samples.append(q_new)
-                initial_potential = final_V
-                initial_potential_grad = final_dVdq
-            else:
-                samples.append(np.copy(samples[-1]))
+        if not diverging and np.random.rand() < p_accept:
```
Comment (ColCarroll) [inline minimc/minimc.py]: this helps me realize that we're doing this all wrong and the divergences should be in the `integrator` function:
1. if `diverging` is true, then the assumption is that we will (almost) never accept
2. so you can remove the `not diverging and` from this block you wrote
3. so the code is the same as it was before

By putting it into the integrator, we can save computation time by bailing out early.

We should also write down when we diverge and when we don't.

**Comment by eigenfoo** [comment]
Superceded (kind of) by #10 😢

**Comment by ColCarroll** [comment]
:cry:

**Pull request closed without merging**